bot/cogs/manage_classes_cog.py

```diff
@@ -1,365 +1,172 @@
-
-            # Finding the category and creating one failed
-        # We cant do anything more, bail out early
-        if not category:
-            embed = discord.Embed(
-                title=f"Error: Category {class_repr.category} not found and not created, Exiting Wizard",
-                color=Colors.Error,
-            )
-            await ctx.send(embed=embed)
+        # validate our given prefix and course number, if given
+        if prefix and not valid_course_maj(prefix):
+            embed = error_embed(inter.user, f'The given course prefix `{prefix}` is invalid.')
+            await inter.response.send_message(embed=embed, ephemeral=True)
             return
+        if course_number and not valid_course_num(course_number):
+            embed = error_embed(inter.user, f'The given course number `{course_number}` is invalid.')
+            await inter.response.send_message(embed=embed, ephemeral=True)
+            return
+        await inter.response.send_modal(AddClassModal(self.bot, class_data=(prefix, course_number)))
 
-        # Create the class channel in the given category
-        channel = await self.create_channel(category, class_repr)
-        await channel.send(f"Here is your generated class channel {ctx.author.mention}, Good luck!")
-
-        # create a class role and mark it as assignable
-        role = await self.create_role(ctx, class_repr)
-
-        # Sleep here to make sure the role has been sent to the database and added
-        await asyncio.sleep(0.5)
-        msg = await ctx.send(f'!role add {role.id}')
-        await msg.delete(delay=1)
-
-        # sync perms with cleanup role
-        await self.sync_perms(ctx, channel, role)
-
-    async def input_class(self, ctx, class_repr: ClassType) -> ClassType | None:
-        def input_check(msg: discord.Message) -> bool:
-            return msg.author == ctx.author and ctx.channel == msg.channel
-
-        # check if the initial command contained a class abbv and number
-        if not class_repr.abbv:
-            embed = discord.Embed(
-                title="**New class setup wizard started :white_check_mark:**",
-                color=Colors.Purple,
-            )
-            avi = ctx.author.display_avatar.url
-            embed.set_footer(text=f"{ctx.author} - Time Limit: {TIMEOUT} Seconds", icon_url=avi)
-            embed.add_field(name="**Current values**", value=class_repr)
-            embed.add_field(
-                name="Please enter the class abbreviation and name E.G.",
-                value="cpsc-1010",
-                inline=False,
-            )
+    @app_commands.command(name='insert', description='Insert a class channel.')
+    async def insert(self, inter: discord.Interaction, channel: discord.TextChannel, role: discord.Role | None = None):
+        # check perms
+        if not await self._perms_check(inter):
+            return
+        # check if a current semester exists
+        if not await self.repo.get_current_semester():
```

Comment:
Yknow code is more readable and looks nicer when spaced out reasonably. General rule I follow is a new line after dedentation occurs.

Reply:
Y'know I've always been the kind of person to keep all of my code compact for some reason, and I can't seem to shake the habit. I was hoping the comments would be a kind of "compromise".